pkg/sql/catalog/descs/factory.go

```diff
@@ -35,18 +37,30 @@ func NewCollectionFactory(
 ) *CollectionFactory {
 	return &CollectionFactory{
 		settings:       settings,
+		codec:          leaseMgr.Codec(),
 		leaseMgr:       leaseMgr,
 		virtualSchemas: virtualSchemas,
 		hydratedTables: hydratedTables,
 	}
 }
 
+// NewBareBonesCollectionFactory constructs a new CollectionFactory which holds
+// onto a minimum of dependencies needed to construct an operable Collection.
+func NewBareBonesCollectionFactory(
+	settings *cluster.Settings, codec keys.SQLCodec,
+) *CollectionFactory {
+	return &CollectionFactory{
+		settings: settings,
+		codec:    codec,
+	}
+}
```

Comment:
This is not just for tests, it's also for builtins and annoying use cases like that. I didn't feel like stuffing the collection factory in the `tree.EvalContext` was a good solution at this stage. Doing it right involves creating interfaces in `tree` which feels somewhat premature.